include/espressif/phy_info.h

```diff
@@ -0,0 +1,456 @@
+*/
+#define DUAL_ANT_CONFIG_NONE 0
+/* antenna diversity for WiFi, use GPIO0 + U0RXD (?) */
+#define DUAL_ANT_CONFIG_DUAL 1
+/* TX/RX switch for external PA & LNA: GPIO 0 high, GPIO 3 low during TX */
+#define DUAL_ANT_CONFIG_TX_GPIO0_HIGH_GPIO3_LOW
+/* TX/RX switch for external PA & LNA: GPIO 0 low, GPIO 3 high during TX */
+#define DUAL_ANT_CONFIG_TX_GPIO0_LOW_GPIO3_HIGH
+
+
+/* CRYSTAL_SLEEP_xx values used for sdk_phy_info_t.crystal_sleep
+ */
+#define CRYSTAL_SLEEP_OFF 0
+#define CRYSTAL_SLEEP_ON 1
+#define CRYSTAL_SLEEP_GPIO16 2
+#define CRYSTAL_SLEEP_GPIO2 3
+
+/* RF Stage 0 & 1 attenuation constants. Use for sdk_phy_info_t.lp_atten_stage01
+
+   @note These values have been tested and are confirmed to work as
+   expected - provided low_power_en is set then the signal is
+   attenuated as per this setting.
+
+   It may look like LP_ATTEN_STAGE01_11_5DB is out of order, but
+   according to monitor mode captures this is the correct ordering of
+   these constants.
+
+   Setting the numeric values in between these constants appears to
+   also attenuate the signal, but not necessarily by the amount you'd
+   expect.
+*/
+#define LP_ATTEN_STAGE01_0DB    0x0f /*     0dB */
+#define LP_ATTEN_STAGE01_2_5DB  0x0e /*  -2.5dB */
+#define LP_ATTEN_STAGE01_6DB    0x0d /*    -6dB */
+#define LP_ATTEN_STAGE01_8_5DB  0x09 /*  -8.5dB */
+#define LP_ATTEN_STAGE01_11_5DB 0x0c /* -11.5dB */
+#define LP_ATTEN_STAGE01_14DB   0x08 /*   -14dB */
+#define LP_ATTEN_STAGE01_17_5DB 0x04 /* -17.5dB */
+#define LP_ATTEN_STAGE01_23DB   0x00 /*   -23dB */
+
+/* Constant for sdk_phy_info_t.pa_vdd */
+#define PA_VDD_MEASURE_VCC 0xFF
+
+/* Bitmask flags for sdk_phy_info_t.freq_correct_mode */
+
+/* Set this flag to disable frequency offset correction */
+#define FREQ_CORRECT_DISABLE 0
+
+/* Set this flag to enable frequency offset correction */
+#define FREQ_CORRECT_ENABLE BIT(0)
+
+/* Set = Baseband PLL frequency is 160MHz (can only apply +ve offset)
+ * Unset = Baseband PLL frequency is 168MHz (can apply +ve/-ve offset */
+#define FREQ_CORRECT_BB_160M BIT(1)
+
+/* Set = use force_freq_offset field to correct, Unset = automatically
+   measure &correct offset
+*/
+#define FREQ_CORRECT_FORCE BIT(2)
+
+
+/* RF_CAL_MODE_xx fields used for sdk_phy_info_t.rf_cal_mode
+ */
+/* Use saved RF CAL data from flash, only. RF init takes 2ms. */
+#define RF_CAL_MODE_SAVED 0
+/* Calibrate TX power control only, use saved RF CAL data for others.
+   RF init takes 20ms. */
+#define RF_CAL_MODE_TXPOWER_ONLY 1
+/* Unclear if/how this mode is different to 2? */
+#define RF_CAL_MODE_SAVED_2 2
+/* Run full RF CAL routine. RF init takes approx 200ms. */
+#define RF_CAL_MODE_FULL 3
+
+/* Data structure that maps to the phy_info configuration block */
+typedef struct __attribute__((packed)) {
+    uint8_t _reserved00[0x05]; /* 0x00 - 0x04 */
+
+    /* This "version" field was set to 5 in the SDK phy_info,
+       and the original SDK startup code checks it is 5 and then loads
```

Comment:
From the versions I looked at, it seems like Espressif added some new (undocumented) fields towards the end but didn't change a lot else. I couldn't see any examples of them changing this number though. :/ It may not even be a version, it's just the field that the RTOS SDK 0.9.9 happens to check for on startup.